### PR TITLE
KSB-19 New fixes in apptester

### PR DESCRIPTION
**Pull request by mariangavlytskiy**

### CODE REVIEW

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

File: scripts/apptester/apptester.py [hunk 1]
```diff
@@ -245,26 +245,29 @@ def build_applications(self, name=None):
...
                         build_app = self.sandboxframe.build_android_java_demo(item['id'])
+                        build_app_result = self.sandboxframe.is_build_successful(item['id'])
 
-                        if self.sandboxframe.is_build_successful(item['id']):
+                        if build_app_result:
                             self.result_matrix[item['name']] = TestStatus.PASSED
                             print 'Building {}:\n{}'.format(item['name'], build_app)
 
-                        elif not self.sandboxframe.is_build_successful(item['id']):
+                        elif not build_app_result:
```
Comment (vadimol) [inline scripts/apptester/apptester.py:270]: There are two options only: True and False.
So:
if build_app_result:
...
else:
...

File: scripts/apptester/apptester.py [hunk 2]
```diff
@@ -245,26 +245,29 @@ def build_applications(self, name=None):
...
                 if 'java' in item['id'] or 'android' in item['id']:
-
-                    if item['destBinaryFile'] == '':
+                    destBinaryFile = item.get('destBinaryFile', None)
+                    if not destBinaryFile:
                         continue
-
                     else:
                         build_app = self.sandboxframe.build_android_java_demo(item['id'])
+                        build_app_result = self.sandboxframe.is_build_successful(item['id'])
```
Comment (vadimol) [inline scripts/apptester/apptester.py:264]: You can receive an exception here. You should catch here.

File: scripts/apptester/apptester.py [hunk 3]
```diff
@@ -274,6 +277,13 @@ def build_android_java_demo(self):
             except Exception as ex:
                 print type(ex), ex
 
+        for app in self.result_matrix:
```
Comment (vadimol) [inline scripts/apptester/apptester.py:280]: This is a dirty hack. Let's discuss a better way to process results

File: scripts/apptester/apptester.py [hunk 4]
```diff
@@ -274,6 +277,13 @@ def build_android_java_demo(self):
             except Exception as ex:
                 print type(ex), ex
```
Comment (vadimol) [inline scripts/apptester/apptester.py:278]: This is redundant here. You should narrow the scope of the exception handling.

**Review by vadimol — CHANGES_REQUESTED** [review]

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Comment by Slash32** [comment]
@kaa-leeroy test this fully please

**Pull request closed without merging**